src/frontend/screens/SendJoinRequestScreen.tsx

```diff
@@ -0,0 +1,121 @@
+import * as React from "react";
+import { Share, StyleSheet, View } from "react-native";
+import { FormattedMessage, defineMessages } from "react-intl";
+import { getUniqueId } from "react-native-device-info";
+import {
+  NavigationStackScreenComponent,
+  TransitionPresets,
+} from "react-navigation-stack";
+
+import useWifiStatus from "../hooks/useWifiStatus";
+import { MEDIUM_BLUE, WHITE } from "../lib/styles";
+import HeaderTitle from "../sharedComponents/HeaderTitle";
+import { BackIcon } from "../sharedComponents/icons";
+import Button from "../sharedComponents/Button";
+import IconButton from "../sharedComponents/IconButton";
+import WifiBar from "../sharedComponents/WifiBar";
+import Text from "../sharedComponents/Text";
+
+const m = defineMessages({
+  title: {
+    id: "screens.SendJoinRequest.title",
+    defaultMessage: "Send Join Request",
+  },
+  verificationCode: {
+    id: "screens.SendJoinRequest.verificationCode",
+    defaultMessage: "Verification Code",
+  },
+  instructions: {
+    id: "screens.SendJoinRequest.instructions",
+    defaultMessage: "Share this code with the Project Coordinator",
+  },
+  share: {
+    id: "screens.SendJoinRequest.share",
+    defaultMessage: "Share",
+  },
+});
+
+const SendJoinRequestScreen: NavigationStackScreenComponent = () => {
+  const { ssid } = useWifiStatus();
+
+  const deviceName = "Android " + getUniqueId().slice(0, 4).toUpperCase();
+
+  // TOOD: Need to properly generate
+  const verificationCode = Math.random().toString().slice(-5);
```

Comment:
@gmaclennan just wanted to highlight this given [this comment](https://github.com/digidem/mapeo-mobile/pull/643#discussion_r676217582) on the original PR. Would love your thoughts/input!